app/controllers/results_controller.rb

```diff
@@ -669,26 +669,13 @@ def get_test_runs_results
         g['created_at_user_name'] = "#{I18n.l(g['created_at'])} (#{User.find(g['user_id']).user_name})"
         g['marks_earned'] = g['test_results.marks_earned']
         g['marks_total'] = g['test_results.marks_total']
+        if g['actual_output'] == 'NA' || g['actual_output'].nil?
+          g['actual_output'] = ''
+        end
       end
       render json: test_script_results
     else
-      result = Result.find(params[:id])
-      if result.released_to_students?
-        test_script_results = TestScriptResult.joins(:test_run, :test_script, :test_results)
-          .where(test_runs: { submission_id: params[:submission_id] })
-          .pluck_to_hash(:created_at, :name, :file_name,
-                  :completion_status,
-                  'test_results.marks_earned', 'test_results.marks_total')
-
-        test_script_results.each do |g|
-          g['created_at_user_name'] = I18n.l(g['created_at'])
-          g['marks_earned'] = g['test_results.marks_earned']
-          g['marks_total'] = g['test_results.marks_total']
-        end
-        render json: test_script_results
-      else
-        head :bad_request
-      end
+      # shouldn't get here
```

Comment:
Don't do this. Instead: 

1. Remove the outer `if` on line 659, and
2. add `:get_test_runs_results` to the list of methods under `authorize_for_admin_and_ta` at the top of this file.